dataprofiler/tests/labelers/test_base_model.py

```diff
@@ -96,25 +96,98 @@ def test_set_parameters(self):
+            base_model.BaseModel.add_label(mock_model, label)
+
+        # assert bad same_as input
+        with self.assertRaisesRegex(TypeError, '`same_as` must be a str.'):
+            base_model.BaseModel.add_label(mock_model, label='test', same_as=1)
+
+        label = 'NEW_LABEL_2'
+        same_as = 'DOES_NOT_EXIST'
         with self.assertRaisesRegex(ValueError,
-                                    "`label_mapping` must be a dict which maps "
-                                    "labels to index encodings."):
+                                    '`same_as` value: {}, did not exist in the '
+                                    'label_mapping.'.format(same_as)):
+            base_model.BaseModel.add_label(mock_model, label, same_as)
+
+        # assert successful add
+        label = 'NEW_LABEL_2'
+        base_model.BaseModel.add_label(mock_model, label)
+        self.assertDictEqual(
+            {'NEW_LABEL': 1, 'NEW_LABEL_2': 2}, mock_model._label_mapping)
+
+        # assert successful add w/ same_as
+        label = 'NEW_LABEL_3'
+        base_model.BaseModel.add_label(mock_model, label, same_as='NEW_LABEL')
+        self.assertDictEqual(
+            {'NEW_LABEL': 1, 'NEW_LABEL_2': 2, 'NEW_LABEL_3': 1},
+            mock_model._label_mapping)
+
+    def test_set_label_mapping_parameters(self):
+
+        # setup mock
+        mock_model = mock.Mock(spec=base_model.BaseModel)
+        mock_model._convert_labels_to_label_mapping.side_effect = \
+            base_model.BaseModel._convert_labels_to_label_mapping
+
+        # assert non value is not accepted.
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
             base_model.BaseModel.set_label_mapping(
                 mock_model, label_mapping=None)
 
+        # non-acceptable value case
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
+            base_model.BaseModel.set_label_mapping(
+                mock_model, label_mapping=1)
+
         # assert error for empty label_mapping dict
-        with self.assertRaisesRegex(ValueError,
-                                    "`label_mapping` must be a dict which maps "
-                                    "labels to index encodings."):
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
             base_model.BaseModel.set_label_mapping(mock_model, label_mapping={})
```

Comment:
can we adjust the message to cope with non-empty dict requirement?

Reply:
Can you explain further what you mean / provide an example?